components/dashboard/src/settings/Preferences.tsx

```diff
@@ -49,13 +50,30 @@ export default function Preferences() {
         }
     };
 
+    const setWorkspaceClass = async (value: string) => {
+        const additionalData = user?.additionalData || {};
+        const prevWorkspaceClass = additionalData?.workspaceClasses?.regular;
+        const workspaceClasses = (additionalData?.workspaceClasses || {}) as WorkspaceClasses;
+        workspaceClasses.regular = value;
+        workspaceClasses.prebuild = value;
+        additionalData.workspaceClasses = workspaceClasses;
+        if (value !== prevWorkspaceClass) {
+            await getGitpodService().server.updateLoggedInUser({ additionalData });
+        }
+        return prevWorkspaceClass;
+    };
+
     return (
         <div>
             <PageWithSettingsSubMenu title="Preferences" subtitle="Configure user preferences.">
                 <h3>Editor</h3>
                 <p className="text-base text-gray-500 dark:text-gray-400">Choose the editor for opening workspaces.</p>
                 <SelectIDE location="preferences" />
-                <SelectWorkspaceClass enabled={BillingMode.canSetWorkspaceClass(userBillingMode)} />
+                <SelectWorkspaceClass
+                    workspaceClass={user?.additionalData?.workspaceClasses?.regular}
+                    enabled={BillingMode.canSetWorkspaceClass(userBillingMode)}
+                    setWorkspaceClass={setWorkspaceClass}
+                />
```

Comment:
**issue**: The duplicate preference in user settings and project settings now poses a question on which preference overrides which. The workspace class scope could be something worth documenting if it's unclear but the project setting should override the user setting, correct?

> **Settings scope for workspace classes**
> Your project setting overrides the user setting.